cache.go

```diff
@@ -240,9 +252,17 @@ func getEncodingStructType(t reflect.Type) (*encodingStructType, error) {
 	var err error
 	var hasKeyAsInt bool
 	var hasKeyAsStr bool
+	var unknownField *field
 	var omitEmptyIdx []int
 	e := getEncodeBuffer()
 	for i := 0; i < len(flds); i++ {
+		if flds[i].unknown {
+			unknownField = flds[i]
+			flds[i] = flds[len(flds)-1]
+			flds = flds[:len(flds)-1]
+			i--
+			continue
```

Comment:
```suggestion
			break
```
Since only one unknown field is allowed?